gateway/api/services/file_storage.py

```diff
@@ -0,0 +1,116 @@
+"""
+This file stores the logic to manage the access to data stores
+"""
+import glob
+import logging
+import os
+
+from django.conf import settings
+
+from utils import sanitize_file_path
+
+USER_STORAGE = "user"
+PROVIDER_STORAGE = "provider"
+
+SUPPORTED_FILE_EXTENSIONS = [".tar", ".h5"]
+
+logger = logging.getLogger("gateway")
+
+
+class FileStorage:  # pylint: disable=too-few-public-methods
+    """
+    The main objective of this class is to manage the access of the users to their storage.
+
+    Attributes:
+        username (str): storgae user's username
+        file_path (str): base user storage file path
+    """
+
+    def __init__(
+        self,
+        username: str,
+        working_dir: str,
+        function_title: str | None,
+        provider_name: str | None,
+    ) -> None:
+        self.file_path = None
+        self.username = username
+
+        if working_dir == USER_STORAGE:
+            self.file_path = self.__get_user_path(
+                username, function_title, provider_name
+            )
+        elif working_dir == PROVIDER_STORAGE:
+            self.file_path = self.__get_provider_path(function_title, provider_name)
+
+    def __get_user_path(
+        self, username: str, function_title: str | None, provider_name: str | None
+    ) -> str:
+        """
+        This method returns the path where the user will store its files
+
+        Args:
+            username (str): username folder where files are going to be stored
+            function_title (str | None): in case the function is from a
+                provider it will identify the function folder
+            provider_name (str | None): in case a provider is provided it will
+                identify the folder for the specific function
+
+        Returns:
+            str: storage path.
+                - In case the working_dir would be a provider that path would
+                    be: username/provider_name/function_title
+                - In case the working_dir would be a user that path would
+                    be: username/
+        """
+        if provider_name is None:
+            path = username
+        else:
+            path = f"{username}/{provider_name}/{function_title}"
+        return os.path.join(
+            sanitize_file_path(settings.MEDIA_ROOT),
+            sanitize_file_path(path),
+        )
```

Comment:
```suggestion
        path = os.path.join(settings.MEDIA_ROOT, username)
        if provider_name:
            path = os.path.join(path, provider_name, function_title)
            
        return sanitize_file_path(path)
```

How about join everyone with the `path.join` and sanitize at the end.